src/ApplicationInsights.ServiceFabric.Native/Net45/CorrelatingServiceRemotingClient.cs

```diff
@@ -0,0 +1,177 @@
+            this.baggageSerializer = new Lazy<DataContractSerializer>(() => new DataContractSerializer(typeof(IEnumerable<KeyValuePair<string, string>>)));
+            this.telemetryClient = new TelemetryClient();
+        }
+
+        /// <summary>
+        /// Gets or Sets the Resolved service partition which was used when this client was created.
+        /// </summary>
+        public ResolvedServicePartition ResolvedServicePartition { get => this.InnerClient.ResolvedServicePartition; set => this.InnerClient.ResolvedServicePartition = value; }
+
+        /// <summary>
+        /// Gets or Sets the name of the listener in the replica or instance to which the client is connected to.
+        /// </summary>
+        public string ListenerName { get => this.InnerClient.ListenerName; set => this.InnerClient.ListenerName = value; }
+
+        /// <summary>
+        /// Gets or Sets the service endpoint to which the client is connected to.
+        /// </summary>
+        public ResolvedServiceEndpoint Endpoint { get => this.InnerClient.Endpoint; set => this.InnerClient.Endpoint = value; }
+
+        /// <summary>
+        /// Gets the inner client which this client wraps.
+        /// </summary>
+        public IServiceRemotingClient InnerClient { get; private set; }
+
+        /// <summary>
+        /// Sends a message to the service and gets a response back. The correlation id and context are sent along with
+        /// the message as message headers.
+        /// </summary>
+        /// <param name="messageHeaders">Message headers.</param>
+        /// <param name="requestBody">Message body.</param>
+        /// <returns>Response body.</returns>
+        public Task<byte[]> RequestResponseAsync(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody)
+        {
+            return SendAndTrackRequestAsync(messageHeaders, requestBody, () => this.InnerClient.RequestResponseAsync(messageHeaders, requestBody));
+        }
+
+        /// <summary>
+        /// Sends a one-way message to the service. The correlation id and context are sent along with the message
+        /// as message headers.
+        /// </summary>
+        /// <param name="messageHeaders">Message headers.</param>
+        /// <param name="requestBody">Message body.</param>
+        public void SendOneWay(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody)
+        {
+            SendAndTrackRequestAsync(messageHeaders, requestBody, () =>
+            {
+                this.InnerClient.SendOneWay(messageHeaders, requestBody);
+                return Task.FromResult<byte[]>(null);
+            }).Forget();
+        }
+
+        private async Task<byte[]> SendAndTrackRequestAsync(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody, Func<Task<byte[]>> doSendRequest)
+        {
+            DependencyTelemetry dt = new DependencyTelemetry();
+            dt.Name = ServiceRemotingLoggingStrings.OutboundRequestActivityName;
```

Comment:
Name for HTTP dependencies contains method + path, e.g. "GET api/default". is there anything in the SF remote call that could allow distinguishing remote calls?